overlay/mkcrate-nobuild.nix

```diff
@@ -0,0 +1,39 @@
+# Creates a derivation for a crate for inputs propagation only.
+{ lib, stdenv }:
+{
+  release, # Compiling in release mode?
+  name,
+  version,
+  registry,
+  src,
+  features,
+  dependencies,
+  devDependencies,
+  buildDependencies,
+  compileMode ? "build",
+  profile,
+  meta ? { },
+  rustcflags ? [ ],
+  rustcBuildFlags ? [ ],
+}:
+with lib; with builtins;
+let
+  inherit
+    (({ right, wrong }: { runtimeDependencies = right; buildtimeDependencies = wrong; })
+      (partition (drv: drv.stdenv.hostPlatform == stdenv.hostPlatform)
+        (concatLists [
+          (attrValues dependencies)
+          (optionals (compileMode == "test") (attrValues devDependencies))
+          (attrValues buildDependencies)
+        ])))
+    runtimeDependencies buildtimeDependencies;
+in stdenv.mkDerivation {
+  name = "crate-${name}-${version}";
+  buildInputs = runtimeDependencies;
+  propagatedBuildInputs = concatMap (drv: drv.propagatedBuildInputs) runtimeDependencies;
+  nativeBuildInputs = buildtimeDependencies;
+  phases = "installPhase fixupPhase";
+  installPhase = "mkdir -p $out";
+  preferLocalBuild = true;
```

Comment:
Yeah, I think farming the crate itself out should be... fine?  The only advantage to doing the rust build locally AFAIK is to populate `/target/` for vanilla cargo tools to re-use in local user workflows.

Reply:
This is a special version of `mkRustCrate` which does nothing but `mkdir -p $out` and creating Nix support files (which is exactly the same as what `propagateEnv` does).